pkgs/servers/http/lighttpd/default.nix

```diff
@@ -20,6 +21,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-4Uidn6dJb78uBxwzi1k7IwDTjCPx5ZZ+UsnvSC4bDiY=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "macos-10.12-avoid-ccrandomgeneratebytes.patch";
+      url = "https://redmine.lighttpd.net/projects/lighttpd/repository/14/revisions/6791f71b20a127b5b0091020dd065f4f9c7cafb6/diff?format=diff";
```

Comment:
For reference, this is the content of the patch being added here:

```
commit 6791f71b20a127b5b0091020dd065f4f9c7cafb6
Author: Glenn Strauss <gstrauss@gluelogic.com>
Date:   Fri Jan 21 13:55:10 2022 -0500

    [core] avoid CCRandomGenerateBytes on MacOS <10.12 (fixes #3140)
    
    (thx ryandesign)
    
    x-ref:
      "lighttpd build failure on macOS 10.13 and earlier"
      https://redmine.lighttpd.net/issues/3140

diff --git a/src/rand.c b/src/rand.c
index caa58bd7..20adbfdd 100644
--- a/src/rand.c
+++ b/src/rand.c
@@ -82,9 +82,10 @@
 #include <sys/ioctl.h>
 #endif
 #if defined(__APPLE__) && defined(__MACH__)
-#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101000 /* OS X 10.10+ */
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101200 /* OS X 10.12+ */
 #undef HAVE_ARC4RANDOM_BUF
 #define HAVE_CCRANDOMGENERATEBYTES
+#include <CommonCrypto/CommonCryptoError.h>
 #include <CommonCrypto/CommonRandom.h>
 #endif
 #endif
```

Looks alright to me, but I wonder whether we should just add the patch into the repository, instead of using `fetchpatch`. I've seen both methods used on different packages, but I'm not sure which is the preferred one.

Reply:
`fetchpatch` is greatly preferred these days. Patches are only kept in-repo either if they've needed modification or if they're in a very early-stage package which can't use `fetchpatch` due to recursion issues.